modules/gdscript/gdscript_warning.cpp

```diff
@@ -229,6 +231,7 @@ String GDScriptWarning::get_name_from_code(Code p_code) {
 		"INTEGER_DIVISION",
 		"NARROWING_CONVERSION",
 		"INT_AS_ENUM_WITHOUT_CAST",
+		"BOOL_AS_ENUM_WITHOUT_CAST",
```

Comment:
So you're suggesting to potentially generalise the `INT_AS_ENUM_WITHOUT_CAST` warning to encompass all 3

Reply:
See https://github.com/godotengine/godot/issues/98830#issuecomment-2457260436. A separate warning for `int` makes sense for historical reasons and because enums are implemented using `int`. However, what we should do with types implicitly convertible to `int` is questionable.